### PR TITLE
Optimizes htp_list_array_replace avoiding useless loop

### PR DESCRIPTION
**Pull request by catenacyber**
Going from 7 to 6 seconds on fuzzing corpus

I am not sure that we need to check for the integer overflow...

### CODE REVIEW

**Comment by catenacyber** [comment]
By the way, this solves the timeout referenced in https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=12354

File: htp/htp_list.c [hunk 1]
```diff
@@ -172,12 +172,15 @@ htp_status_t htp_list_array_replace(htp_list_array_t *l, size_t idx, void *e) {
...
     if (idx + 1 > l->current_size) return HTP_DECLINED;
 
-    size_t i = l->first;
-
-    while (idx--) {
-        if (++i == l->max_size) {
-            i = 0;
-        }
+    size_t i;
+    // Do we need to check for this integer overflow ?
```
Comment (victorjulien) [inline htp/htp_list.c]: not sure actually, size_t is 32bit on 32bit machines. To get to 4 billion transactions (the only place this code is used) we'd need a lot of traffic. But then again it's probably not completely impossible. From a quick look at the rest of the list code I don't get the impression a overflow is really handled in other places. Would you agree?

Comment (catenacyber) [reply htp/htp_list.c]: Indeed, it is not checked in `htp_list_array_get` where there is the same addition `l->first + idx`

**Comment by victorjulien** [comment]
Can you do a new PR a single commit? I don't think a PR should have commits fixing other commits in the same PR. Thanks!

**Pull request closed without merging**